src/FeedRegistration.php

```diff
@@ -0,0 +1,239 @@
+<?php
+/**
+ * Pinterest for WooCommerce Feed Registration.
+ *
+ * @package     Pinterest_For_WooCommerce/Classes/
+ * @version     x.x.x
+ */
+
+namespace Automattic\WooCommerce\Pinterest;
+
+use Automattic\WooCommerce\Pinterest\Utilities\FeedLogger;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Handling feed files registration.
+ */
+class FeedRegistration {
+
+	use FeedLogger;
+
+	const ACTION_HANDLE_FEED_REGISTRATION = PINTEREST_FOR_WOOCOMMERCE_PREFIX . '-handle-feed-registration';
+
+	/**
+	 * Local Feed Configurations class.
+	 *
+	 * @var LocalFeedConfigs of local feed configurations;
+	 */
+	private $configurations;
+
+	/**
+	 * Feed File Generator Instance
+	 *
+	 * @var $feed_generator FeedGenerator
+	 */
+	private $feed_generator = null;
+
+	/**
+	 * Feed Registration.
+	 *
+	 * @since x.x.x
+	 * @param LocalFeedConfigs $local_feeds_configurations Locations configuration class.
+	 * @param FeedGenerator    $feed_generator Feed generator class.
+	 */
+	public function __construct( $local_feeds_configurations, $feed_generator ) {
+		$this->configurations = $local_feeds_configurations;
+		$this->feed_generator = $feed_generator;
+
+		add_action( self::ACTION_HANDLE_FEED_REGISTRATION, array( $this, 'handle_feed_registration' ) );
+		if ( false === as_has_scheduled_action( self::ACTION_HANDLE_FEED_REGISTRATION, array(), PINTEREST_FOR_WOOCOMMERCE_PREFIX ) ) {
+			as_schedule_recurring_action( time() + 10, 10 * MINUTE_IN_SECONDS, self::ACTION_HANDLE_FEED_REGISTRATION, array(), PINTEREST_FOR_WOOCOMMERCE_PREFIX );
+		}
+	}
+
+	/**
+	 * Check if the feed is registered based on the plugin's settings.
+	 * If not, try to register it,
+	 * Log issues.
+	 * Potentially report issues.
+	 *
+	 * Should be run on demand when settings change,
+	 * and on a scheduled basis.
+	 *
+	 * @return mixed
+	 *
+	 * @throws \Exception PHP Exception.
+	 */
+	public function handle_feed_registration() {
+
+		if ( ! $this->feed_file_exists() ) {
+			self::log( 'Feed didn\'t fully generate yet. Retrying later.', 'debug' );
```

Comment:
We should switch to double quotes here to avoid using a `\` in the string.

```suggestion
			self::log( "Feed didn't fully generate yet. Retrying later.", 'debug' );
```

Reply:
Yep.